Tests/tests_end_to_end/content/xsoar_saas/test_e2e_xsoar_saas.py

```diff
@@ -0,0 +1,226 @@
+            username = integration_params["credentials"]["identifier"]
+            password = integration_params["credentials"]["password"]
+            headers = {"Accept": "application/taxii+json;version=2.1"}
+            response = xsoar_saas_client.do_long_running_instance_request(
+                instance_name,
+                url_suffix="/threatintel/collections",
+                headers=headers,
+                username=username,
+                password=password
+            )
+
+            # get the collections available
+            collections = get_json_response(response).get("collections")
+            assert collections, f'could not get collections from url={response.request.url}, ' \
+                f'status_code={response.status_code}, response={collections}'
+
+            collection_id = collections[0]["id"]
+
+            # get the actual indicators from the collection
+            response = xsoar_saas_client.do_long_running_instance_request(
+                instance_name,
+                url_suffix=f"/threatintel/collections/{collection_id}/objects",
+                headers=headers,
+                username=username,
+                password=password
+            )
+
+            indicators = get_json_response(response).get("objects")
+            assert indicators, f'could not get indicators from url={response.request.url} with available ' \
+                f'indicators={available_indicators}, status code={response.status_code}, response={indicators}'
+
+
+def test_slack_ask(request: SubRequest, xsoar_saas_client: XsoarSaasClient):
+    """
+    Given:
+        - playbook that runs slack-ask (runs SlackAskV2 and then answers the slack ask with the slack V3 integration)
+        - slack V3 integration
+    When:
+        - running slack ask flow with slack V3
+    Then:
+        - make sure that the slack ask flow worked properly
+        - make sure that the playbook finishes.
+        - make sure that the context is populated with thread ID(s) from the slack-ask and slack-response.
+    """
+    integration_params = get_integration_params(
+        request.config.option.integration_secrets_path, instance_name="slack-e2e-instance"
+    )
+
+    with save_integration_instance(
+        xsoar_saas_client,
+        integration_params=integration_params,
+        integration_id="SlackV3",
+        is_long_running=True,
+        instance_name=get_integration_instance_name(integration_params, default="SlackV3")
+    ):
+        playbook_id_name = "TestSlackAskE2E"
+        with save_playbook(
+            xsoar_saas_client,
+            playbook_path="Tests/tests_end_to_end/content/xsoar_saas/TestSlackAskE2E.yml",
+            playbook_id=playbook_id_name,
+            playbook_name=playbook_id_name
+        ), save_incident(xsoar_saas_client, playbook_id=playbook_id_name) as incident_response:
+            # make sure the playbook finished successfully
+            assert is_playbook_state_as_expected(
```

Comment:
we dont need to poll/add a sleep here?

Reply:
see in `client_utils.py` it is implemented with the `retry` decorator from the sdk.

We need to keep polling instead of just sleeping for X seconds.